src/main/java/org/truffleruby/RubyContext.java

```diff
@@ -257,7 +267,99 @@ public void initialize() {
             sharedObjects.startSharing();
         }
 
+        if (isPreInitializing()) {
+            // Cannot save the FileDescriptor in the image, referenced by the SecureRandom instance
+            random = null;
+            // Cannot save native functions in the image
+            truffleNFIPlatform = null;
```

Comment:
Isn't this already set to null?

Reply:
Yes. I will remove it since it is now redundant.
In early experiments it was possible to use NFI during context pre-initialization, so it was only null-ed here. But currently it is no longer possible to use NFI during context pre-initialization, so NFI is always null during context pre-initialization.